packages/samples/headless-react/src/components/triggers/notify-trigger.class.tsx

```diff
@@ -0,0 +1,38 @@
+import {Component, ContextType} from 'react';
+import {
+  buildNotifyTrigger,
+  NotifyTrigger as HeadlessNotifyTrigger,
+  NotifyTriggerState,
+  Unsubscribe,
+} from '@coveo/headless';
+import {AppContext} from '../../context/engine';
+
+export class NotifyTrigger extends Component<{}, NotifyTriggerState> {
+  static contextType = AppContext;
+  context!: ContextType<typeof AppContext>;
+
+  private controller!: HeadlessNotifyTrigger;
+  private unsubscribe: Unsubscribe = () => {};
+
+  componentDidMount() {
+    this.controller = buildNotifyTrigger(this.context.engine!);
+    this.updateState();
+    this.unsubscribe = this.controller.subscribe(() => this.updateState());
+  }
+
+  componentWillUnmount() {
+    this.unsubscribe();
+  }
+
+  private updateState() {
+    this.setState(this.controller.state, () => {
+      if (this.state.notification) {
```

Comment:
Is the condition necessary here if you already check if the string is not empty in the controller's subscribe method?

Reply:
Without this condition, a notification with an empty string is fired on the initial page load since we call updateState() once in ComponentDidMount() to initialize it.